celery/schedules.py

```diff
@@ -101,6 +101,11 @@ def to_local(self, dt):
             return timezone.to_local_fallback(dt)
         return dt
 
+    def __eq__(self, other):
+        if isinstance(other, BaseSchedule):
+            return other.nowfun == self.nowfun
```

Comment:
You are right, if it is re-instantiated by another process, it will not be the same object. However, how will equality check work for functions in general? I am not aware of any concrete method of performing this check. See also [here](http://stackoverflow.com/questions/20059011/check-if-two-python-functions-are-equal).